src/macros.rs

```diff
@@ -1,10 +1,305 @@
+//! Extra macro documentation.
+
 #[doc(hidden)]
 pub use fizyr_rpc_macros::interface as interface_impl;
 
 #[macro_export]
 /// Define an RPC interface.
+///
+/// This macro generates a module with a `Client` and `Server` struct,
+/// and some more helper types.
+///
+/// The client struct can be used to initiate requests and send stream messages.
+/// It can be created from a [`PeerWriteHandle`] or a [`PeerHandle`].
+/// Note that if you create the client from a [`PeerHandle`], the [`PeerReadHandle`] will not be accessible.
+///
+/// For the server struct it is exacatly opposite: it can be used to receive requests and stream messages.
+/// It can be created from a [`PeerReadHandle`] or a [`PeerHandle`],
+/// but creating it from a [`PeerHandle`] will discard the [`PeerWriteHandle`].
+///
+/// # Example
+///
+/// See the [`interface_example`] module for an example, with the source code and generated documentation.
+///
+/// # Syntax
+///
+/// The syntax for the macro is as follows:
+/// ```rust
+/// fizyr_rpc::interface! {
+///     // The `interface` keyword defines an RPC interface.
+///     // You must have exactly one interface definition in the macro invocation.
+///     //
+///     // The macro generates a module with the same name containing the generated types.
+///     // You can adjust the visiblity of the generated module with the `pub` keyword as normal.
+///     // You can use this to take full control over the public module structure,
+///     // by declaring the interface private and re-exporting the module contents from a different location.
+///     //
+///     // Each item in can have user written documentation.
+///     // Simply write doc comments with triple slashes as usual.
+///     // This applies to the interface definition, a service definitions, update definitions and stream definitions.
+///     pub interface $interface_name {
+///         // The `service` keyword defines a service.
+///         //
+///         // You can have any amount of service definition inside an interface definition.
+///         //
+///         // The $id is used as the service ID and must be an i32.
+///         // The ID must be unique for all services in the interface.
```

Comment:
What happens if I make a service with non-unique ID?

Reply:
You'll get a very beautiful error :) Try it :D